src/Megamix/Commands.cpp

```diff
@@ -0,0 +1,112 @@
+#include <3ds.h>
+#include <CTRPluginFramework.hpp>
+
+#include "Megamix.hpp"
+
+using CTRPluginFramework::OSD;
+using CTRPluginFramework::Utils;
+
+namespace Megamix{
+
+    void tickflowCommandsHookWrapper() {
+        asm(
+            "mov r0, r6\n"
+            "bl tickflowCommandsHook\n"
+            "bx r0\n"
+        );
+    }
+
+    extern "C" __attribute__((used)) 
+    int tickflowCommandsHook(CTickflow* self, u32 cmd_num, u32 arg0, u32* args){
+        switch(cmd_num){
+            case 0: // Necessary, as it doesn't execute otherwise.
+                return Region::TickflowAsyncSubLocation();
+            case InputCheck:
+                input_cmd(self, arg0, args);
+            break;
+            case VersionNumber:
+                versionCheck(self, arg0, args);
+            break;
+            case LanguageCheck:
+                languageCheck(self, arg0, args);
+            break;
+            case DisplayCondvar:
+                displayCondvar(self, arg0, args);
+            break;
+        }
+        return Region::TickflowCommandsEnd();
+    }
+
+    void input_cmd(CTickflow* self, u32 arg0, u32* args) {
+        if (arg0 > 2) {
+            self->condvar = 0;
+            return;
+        } else if (arg0 == 2) {
+            CSaveData** gSaveData = (CSaveData**)Region::GlobalSaveDataPointer();
+            arg0 = (u32)(*gSaveData)->fileData[(*gSaveData)->currentFile].playStyle;
+        }
+
+        CInputManager** gInputManager = (CInputManager**)Region::GlobalInputManagerPointer();
+        if (arg0 == 0) {
+            if (args[0] >= 32) {
+                self->condvar = 0;
+                return;
+            }
+            
+            self->condvar = ((u32)(*gInputManager)->padHandler->holdButtons >> args[0]) & 1;
+        } else {
+            self->condvar = ((*gInputManager)->touchPanelHandler->touchPanelStatus.touch);
+        }
+    }
+
+    void versionCheck(CTickflow* self, u32 arg0, u32* args) {
+        if (arg0 == 0){ //RHMPatch version
+            self->condvar = SUPPORTED_RHMPATCH;
+        } else if (arg0 == 1){ //Saltwater version
+            self->condvar = VERSION_MAJOR*0x10
+                +VERSION_MINOR;
+        }
+    }
+
+    void languageCheck(CTickflow* self, u32 arg0, u32* args) {
+        CSaveData** gSaveData = (CSaveData**)Region::GlobalSaveDataPointer();
+        CFileManager** gFileManager = (CFileManager**)Region::GlobalFileManagerPointer();
+        int locale = (*gSaveData)->fileData[(*gSaveData)->currentFile].locale;
+        if(locale == 1){
+            self->condvar = 0;
+        } else {
+            wchar_t sublocale[4];
+            utf16_to_utf32((u32*)sublocale, (*gFileManager)->sublocale, 4);
```

Comment:
the reason you may be having issues here is because you set the size of the wchar_t array to be exactly 4 - which means that it never finds a null terminator. set this to the game's original 9 (or hell even just 5 if you want) and you should avoid needing to use `find`

Reply:
I prefer to keep the `find` to cover both EUEN and USEN, as well as just in case USFR or USSP somehow starts existing

Reply:
fair enough!